Sources/SourceKitLSP/Swift/MacroExpansionReferenceDocumentURLData.swift

```diff
@@ -15,41 +15,71 @@ import LanguageServerProtocol
 import RegexBuilder
 
 /// Represents url of macro expansion reference document as follows:
-/// `sourcekit-lsp://swift-macro-expansion/LaCb-LcCd.swift?primaryFilePath=&fromLine=&fromColumn=&toLine=&toColumn=&bufferName=`
+/// `sourcekit-lsp://swift-macro-expansion/LaCb-LcCd.swift?parent=&fromLine=&fromColumn=&toLine=&toColumn=&bufferName=`
 ///
 /// Here,
 ///  - `LaCb-LcCd.swift`, the `displayName`, represents where the macro will expand to or
 ///    replace in the source file (i.e. `macroExpansionEditRange`)
-///  - `primaryFilePath` denoting the URL of the source file
+///  - `parent` denoting the URI of the document from which the macro was expanded. For a first-level macro expansion,
+///    this is a file URI. For nested macro expansions, this is a `sourcekit-lsp://swift-macro-expansion` URL.
 ///  - `fromLine`, `fromColumn`, `toLine`, `toColumn` represents the cursor's `selectionRange`
 ///  - `bufferName` denotes the buffer name of the specific macro expansion edit
 package struct MacroExpansionReferenceDocumentURLData {
   package static let documentType = "swift-macro-expansion"
 
-  package var primaryFileURL: URL
+  /// The document from which this macro was expanded. For first-level macro expansions, this is a file URL. For
+  /// second-level macro expansions, this is a `sourcekit-lsp://swift-macro-expansion/` URL, third-level macro
+  /// expansions are a `sourcekit-lsp:` URL that themselves have a `sourcekit-lsp:` URL as their parent.
+  package var parent: DocumentURI
+
+  /// The range that was selected in `parent` when the macro was expanded.
   package var selectionRange: Range<Position>
+
+  /// ## Example
+  ///
+  /// User's source File:
+  /// URL: `file:///path/to/swift_file.swift`
+  /// ```swift
+  /// let a = 10
+  /// let b = 5
+  /// print(#stringify(a + b))
+  /// ```
+  ///
+  /// Generated content of reference document url:
+  /// URL:
+  /// `sourcekit-lsp://swift-macro-expansion/L3C7-L3C23.swift?primaryFilePath=/path/to/swift_file.swift&fromLine=3&fromColumn=8&toLine=3&toColumn=8&bufferName=@__swift_macro_..._Stringify_.swift`
```

Comment:
I do notice `primaryFilePath` should be `parent` in doc comment similarly in a few other places. I shall update this in my PR.